src/main/java/com/example/Trello/services/impl/CardServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.example.Trello.services.impl;
+
+import com.example.Trello.mappers.CardMapper;
+import com.example.Trello.model.entity.Board;
+import com.example.Trello.model.entity.Card;
+import com.example.Trello.model.dto.card.CardCreation;
+import com.example.Trello.repositories.CardRepository;
+import com.example.Trello.services.BoardService;
+import com.example.Trello.services.CardService;
+import com.example.Trello.services.exception.NoCardFoundException;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@Slf4j
+public class CardServiceImpl implements CardService {
+
+    private static final String GET_CARD_BY_ID_ERROR = "Card with given id doesn't exist";
+
+    private final CardMapper cardMapper;
+
+    private final CardRepository cardRepository;
+
+    private final BoardService boardService;
+
+    public CardServiceImpl(final CardMapper cardMapper, final CardRepository cardRepository, final BoardService boardService) {
+        this.cardMapper = cardMapper;
+        this.cardRepository = cardRepository;
+        this.boardService = boardService;
+    }
+
+    @Override
+    public void addCard(final long id, final CardCreation cardCreation) {
+        final Card card = cardMapper.map(cardCreation);
+
+        card.setBoard(new Board(id));
+        cardRepository.save(card);
+    }
+
+    @Override
+    public List<Card> getCards(final long id) {
+        return cardRepository.findAll()
```

Comment:
you need to get already filtered cards from the DB. Use custom repository methods.
Otherwise in production it will kill your system. Imagine you have 10 mln of cards. But each board has only 5 cards. Instead of getting these 5 cards, you get 10 mln and then start iterating and filtering it.